src/operators/cross.ml

```diff
@@ -131,47 +134,26 @@ class cross ~kind (s : source) ~cross_length ~override_duration ~rms_width
       self#cleanup_transition_source
 
     method private set_clock =
-      let child_clock = Clock.create_known (new Clock.clock self#id) in
-      (* Our external clock should strictly contain the child clock. *)
-      Clock.unify self#clock
-        (Clock.create_unknown ~sources:[] ~sub_clocks:[child_clock]);
-
-      (* The source must belong to our clock, since we need occasional
-       * control on its flow (to fold an end of track over a beginning).
-       *
-       * To be safe, also require that the transition's sources belong
-       * to that same clock, since we (dynamically) activate them as well.
-       * The activation mechanism is designed for a single thread,
-       * which multiples clocks wouldn't ensure. *)
-      Clock.unify child_clock s#clock;
-      Lang.iter_sources (fun s -> Clock.unify child_clock s#clock) transition;
-
-      (* Make sure the child clock can be garbage collected, cf. cue_cut(). *)
-      Gc.finalise (finalise_child_clock child_clock) self
+      child_support#set_clock;
+      Lang.iter_sources
+        (fun s -> Clock.unify self#child_clock s#clock)
+        transition
```

Comment:
This looks like a "big change", can you explain it further? Are you sure that there are no potential side effects?

Reply:
This is actually just a factorization. All the code that was there is now in `Child_support`. I also added some thoughts about what was done and why in `clock.ml`.